### PR TITLE
add SQL Database Projects integration docs

### LINKED ISSUE #2038: [Community Toolkit New article]: SQL Databases Projects
### Proposed topic or title

SQL Databases Projects

### Reason for the article

We've had a contribution for SQL Databases Projects hosting integration for Community Toolkit.

### Article abstract

Overview of using SQL Databases Projects hosting integration.

### Relevant searches

_No response_

### PR DESCRIPTION
**Pull request by ErikEJ**
## Summary

Add docs for SQL database projects integration

Fixes #2038  

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/community-toolkit/hosting-sql-database-projects.md](https://github.com/dotnet/docs-aspire/blob/72605224cb052ea217b425cac0c9fc688a40178a/docs/community-toolkit/hosting-sql-database-projects.md) | [.NET Aspire SQL Database Projects hosting integration](https://review.learn.microsoft.com/en-us/dotnet/aspire/community-toolkit/hosting-sql-database-projects?branch=pr-en-us-2040) |
| [docs/fundamentals/integrations-overview.md](https://github.com/dotnet/docs-aspire/blob/72605224cb052ea217b425cac0c9fc688a40178a/docs/fundamentals/integrations-overview.md) | [.NET Aspire integrations overview](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/integrations-overview?branch=pr-en-us-2040) |
| [docs/toc.yml](https://github.com/dotnet/docs-aspire/blob/72605224cb052ea217b425cac0c9fc688a40178a/docs/toc.yml) | [docs/toc](https://review.learn.microsoft.com/en-us/dotnet/aspire/toc?branch=pr-en-us-2040) |

### CODE REVIEW

**Comment by ErikEJ** [comment]
@Alirexaa @jmezach @aaronpowell  FYI

File: docs/community-toolkit/hosting-sql-projects.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+[!INCLUDE [includes-hosting](../includes/includes-hosting.md)]
+
+[!INCLUDE [banner](includes/banner.md)]
+
+In this article, you learn how to use the .NET Aspire SQL Database Projects hosting integration to publish your database schema to your SQL Server database.
+
+## Prerequisites
+
+This integration requires a SQL Database Project based on either [MSBuild.Sdk.SqlProj](https://github.com/rr-wfm/MSBuild.Sdk.SqlProj) and [Microsoft.Build.Sql](https://github.com/microsoft/DacFx).
```
Comment (jmezach) [inline docs/community-toolkit/hosting-sql-projects.md:18]: `and` should be `or`

File: docs/toc.yml [hunk 2]
```diff
@@ -248,6 +248,8 @@ items:
             href: community-toolkit/ollama.md
           - name: Meilisearch
             href: community-toolkit/hosting-meilisearch.md
+          - name: SQL Database Projects
+            href: community-toolkit/hosting-sql-projects.md
```
Comment (Alirexaa) [inline docs/toc.yml:252]:
```suggestion
            href: community-toolkit/hosting-sql-database-projects.md
```
We used this URL in the community toolkit repo readme.

**Review by Alirexaa — COMMENTED** [review]
Could you make sure to update https://github.com/dotnet/docs-aspire/blob/main/docs/fundamentals/integrations-overview.md?

**Comment by ErikEJ** [comment]
@Alirexaa Done

File: docs/community-toolkit/hosting-sql-database-projects.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+## Example usage
+
+Add a reference to the MSBuild.Sdk.SqlProj or Microsoft.Build.Sql project you want to publish in your .NET Aspire AppHost project:
+
+```dotnetcli
+dotnet add reference ../MySqlProj/MySqlProj.csproj
+```
+
+> Note: Adding this reference will currently result in warning ASPIRE004. This is a known issue and will be resolved in a future release.
```
Comment (aaronpowell) [inline docs/community-toolkit/hosting-sql-database-projects.md:49]:
```suggestion
> [!NOTE]
> Adding this reference will currently result in warning ASPIRE004 on the project due to how references are parsed. The .NET Aspire team is aware of this and we're working on a cleaner solution.
```

File: docs/community-toolkit/hosting-sql-database-projects.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+builder.AddSqlProject("mysqlproj")
+       .WithDacpac("path/to/mysqlproj.dacpac")
+       .WithReference(sql);
+```
+
+### Redeploy support
+
+If you make changes to your SQL Database project while the AppHost is running, you can use the `Redeploy` custom action on the Aspire Dashboard to redeploy your updates without having to restart the AppHost.
+
+## See also
```
Comment (aaronpowell) [inline docs/community-toolkit/hosting-sql-database-projects.md:84]: Should there be a link to the database projects github repo?

**Comment by ErikEJ** [comment]
@aaronpowell Comments addressed, I do not understand the failing build check

**Comment by aaronpowell** [comment]
> @aaronpowell Comments addressed, I do not understand the failing build check

It's fixed. There was a bad link but one of my suggestions removed that link and it's all green now.

**Comment by ErikEJ** [comment]
@aaronpowell Ouch, got the bad link from one of the other docs files 😆

File: docs/community-toolkit/hosting-sql-database-projects.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+## Example usage
+
+Add a reference to the MSBuild.Sdk.SqlProj or Microsoft.Build.Sql project you want to publish in your .NET Aspire AppHost project:
+
+```dotnetcli
+dotnet add reference ../MySqlProj/MySqlProj.csproj
+```
+
+> [!NOTE]
+> Adding this reference will currently result in warning ASPIRE004 on the project due to how references are parsed. The .NET Aspire team is aware of this and we're working on a cleaner solution.
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-sql-database-projects.md:50]:
```suggestion
> Adding this reference will currently result in warning `ASPIRE004` on the project due to how references are parsed. The .NET Aspire team is aware of this and we're working on a cleaner solution.
```

File: docs/community-toolkit/hosting-sql-database-projects.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                  Version="*" />
+```
+
+---
+
+For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).
+
+## Example usage
+
+Add a reference to the MSBuild.Sdk.SqlProj or Microsoft.Build.Sql project you want to publish in your .NET Aspire AppHost project:
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-sql-database-projects.md:43]:
```suggestion
Add a reference to the [📦 MSBuild.Sdk.SqlProj](https://www.nuget.org/packages/MSBuild.Sdk.SqlProj) or [📦 Microsoft.Build.Sql](https://www.nuget.org/packages/Microsoft.Build.Sql) project you want to publish in your .NET Aspire app host project:
```

File: docs/community-toolkit/hosting-sql-database-projects.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+var builder = DistributedApplication.CreateBuilder(args);
+
+var sql = builder.AddSqlServer("sql")
+                 .AddDatabase("test");
+
+builder.AddSqlProject<Projects.MySqlProj>("mysqlproj")
+       .WithReference(sql);
+```
+
+Now when you run your .NET Aspire AppHost project you will see the SQL Database Project being published to the specified SQL Server.
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-sql-database-projects.md:64]:
```suggestion
Now when you run your .NET Aspire app host project you see the SQL Database Project being published to the specified SQL Server.
```

File: docs/community-toolkit/hosting-sql-database-projects.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+### Local .dacpac file support
+
+If you are sourcing your .dacpac file from somewhere other than a project reference, you can also specify the path to the .dacpac file directly:
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-sql-database-projects.md:68]:
```suggestion
If you are sourcing your _.dacpac_ file from somewhere other than a project reference, you can also specify the path to the _.dacpac_ file directly:
```

File: docs/community-toolkit/hosting-sql-database-projects.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                 .AddDatabase("test");
+
+builder.AddSqlProject("mysqlproj")
+       .WithDacpac("path/to/mysqlproj.dacpac")
+       .WithReference(sql);
+```
+
+### Redeploy support
+
+If you make changes to your SQL Database project while the AppHost is running, you can use the `Redeploy` custom action on the Aspire Dashboard to redeploy your updates without having to restart the AppHost.
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-sql-database-projects.md:83]:
```suggestion
If you make changes to your SQL Database project while the app host is running, you can use the `Redeploy` custom action on the .NET Aspire dashboard to redeploy your updates without having to restart the app host.
```

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**